### PR TITLE
Support C# 14 null-conditional assignments

### PR DESCRIPTION
**Pull request by Jim8y**
## Summary
- support C# 14 null-conditional assignments for member, array element, and indexer-property targets
- guard every receiver in a `?.` or `?[]` chain before evaluating indexes or right-hand operands
- support simple assignments, compound assignments, and `??=` while returning `null` when a receiver is skipped
- add runtime tests for null receivers, index and RHS side effects, nested receivers, arrays, multidimensional arrays, and indexers

## Validation
- `dotnet test tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj --filter FullyQualifiedName~UnitTest_NullConditionalAssignment --no-restore`
- `dotnet test tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj --no-restore`

### CODE REVIEW

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs [hunk 1]
```diff
@@ -407,4 +456,160 @@ private void ConvertConditionalMemberBindingAssignment(SemanticModel model, Memb
...
+    private byte[] StoreConditionalElementIndices(SemanticModel model, ElementBindingExpressionSyntax elementBinding, ITypeSymbol receiverType)
+    {
+        IArrayTypeSymbol arrayType = GetConditionalElementArrayType(elementBinding, receiverType);
+        if (arrayType.Rank > 1)
+        {
+            EnsureMultiDimensionalArguments(elementBinding.ArgumentList.Arguments, arrayType.Rank, elementBinding.ArgumentList);
+        }
+        else if (elementBinding.ArgumentList.Arguments.Count != 1)
+        {
+            throw new CompilationException(elementBinding.ArgumentList, DiagnosticId.MultidimensionalArray, $"Unsupported array rank: {elementBinding.ArgumentList.Arguments}");
```
Comment (ajara87) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs]: should not be `elementBinding.ArgumentList.Arguments.Count` ?

Comment (shargon) [reply src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs]: At least `elementBinding.ArgumentList.Arguments`

**Comment by Wi1l-B0t** [comment]
Good.
But maybe complicated.

File: tests/Neo.Compiler.CSharp.TestContracts/Contract_NullConditional.cs [hunk 2]
```diff
@@ -9,6 +9,61 @@ private class Node
...
+
+    private static Node CreateNodeWithSideEffect()
+    {
+        SideEffects += 1;
+        return new Node();
+    }
+
+    private static int CreateValueWithSideEffect()
+    {
+        SideEffects += 1;
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.TestContracts/Contract_NullConditional.cs:59]: How did you check SideEffects?

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.TestContracts/Contract_NullConditional.cs:59]: It is checked through the contract return values in `UnitTest_NullConditionalAssignment`. Each side-effect contract method resets `SideEffects`, runs the null-conditional assignment, then returns the counter or an encoded value that includes it. The tests cover both branches, for example `AssignChildSideEffects(true) == 1` and `AssignChildSideEffects(false) == 0`, plus element/index/value/coalesce/indexer cases such as `AssignElementIndexAndValueSideEffects` and `CoalesceElementIndexAndValueSideEffects`.

Comment (shargon) [reply tests/Neo.Compiler.CSharp.TestContracts/Contract_NullConditional.cs:59]: But if is not stored, every call reset the value

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.TestContracts/Contract_NullConditional.cs:59]: Yes, static fields are reset between contract invocations. The tests do not rely on persistence across calls. `SideEffects` is only a per-invocation probe: each method resets it, runs one null-conditional assignment, and returns the counter in the same invocation. The encoded assertions such as `AssignElementIndexAndValueSideEffects(true) == 27` and `CoalesceElementIndexAndValueSideEffects(true, true) == 13` verify how many side-effect expressions ran within that single call.

**Review by shargon — DISMISSED** [review]

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs [hunk 3]
```diff
@@ -409,4 +463,262 @@ private void ConvertConditionalMemberBindingAssignment(SemanticModel model, Memb
...
+                if (!property.IsStatic) AccessSlot(OpCode.LDLOC, receiverSlot);
+                CallMethodWithConvention(model, property.GetMethod);
+                ConvertExpression(model, right);
+                EmitComplexAssignmentOperator(type, operatorToken);
+                byte propertyValueSlot = AddAnonymousVariable();
+                AccessSlot(OpCode.STLOC, propertyValueSlot);
+                AccessSlot(OpCode.LDLOC, propertyValueSlot);
+                if (!property.IsStatic) AccessSlot(OpCode.LDLOC, receiverSlot);
+                CallMethodWithConvention(model, property.SetMethod, CallingConvention.Cdecl);
+                AccessSlot(OpCode.LDLOC, propertyValueSlot);
```
Comment (ajara87) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs]: Could replacing STLOC + 2× LDLOC with a single DUP work here? One copy stays on the stack as the return value and the setter consumes the other

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs]: Applied. The conditional property compound assignment now keeps the assigned value on the stack with `DUP` before invoking the setter, so it no longer needs the temporary local.

**Review by ajara87 — APPROVED** [review]

**Review by shargon — APPROVED** [review]
